File: lib/commands/figlet.rb
# frozen_string_literal: true

module Rebbot
  module Commands
    class Figlet < Rebbot::Commands::Base
      on :figlet, description: 'runs figlet on input text'

      def with_options(cmd)
        cmd.string('message', 'Message to figlet-ify', required: true)
      end

      def on_event(event)
        event.respond(content: "```\n#{Artii::Base.new.asciify event.options['message'][0..20]}```")
      end
    end
  end
end

File: src/commands/figlet.ts
import { InteractionResponseType } from "discord-interactions";
import figlet from "figlet";
import { CommandOptionType, cmd } from "../types.js";

export default cmd(
  {
    name: "figlet",
    description: "Runs figlet on input text",
    options: [
      {
        name: "message",
        description: "Text to render",
        type: CommandOptionType.STRING,
        required: true,
        max_length: 21,
      },
    ],
  },
  async ({ message }) => {
    const art = await figlet.text(message);
    return {
      type: InteractionResponseType.CHANNEL_MESSAGE_WITH_SOURCE,
      data: { content: "```\n" + art + "\n```" },
    };
  },
);
